src/spec.c

```diff
@@ -1839,17 +1839,20 @@ int IndexSpec_UpdateWithHash(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleSt
   RedisSearchCtx sctx = SEARCH_CTX_STATIC(ctx, spec);
   Document doc = {0};
   Document_Init(&doc, key, 1.0, DEFAULT_LANGUAGE);
-  if (Document_LoadSchemaFields(&doc, &sctx) != REDISMODULE_OK) {
+  // if a key does not exit, is not a hash or has no fields in index schema
+  if (Document_LoadSchemaFields(&doc, &sctx) != REDISMODULE_OK || doc.numFields == 0) {
```

Comment:
@MeirShpilraien 
If the key does not exist, it isn't a hash or has no field we return here and not index

Reply:
Yes but if the key is a hash but all the fields do not match the schema then it will still be added to the doctable no? So we did not break anything?

Reply:
We don't have to break anything but I think it is a good opportunity to do so.
IMO, the current state wastes doctable space and can be reached only by searching for `*`.

Reply:
I agree, @K-Jo @gkorland what do you think?